### PR TITLE
feat: user card actions

### LINKED ISSUE #599: [FEATURE]: User card actions
## Summary

As a super admin the switch button should be visible and editable. This action makes a request to the backend in order to change the user role in the app. The edit button and when the user clicks in the user's name should redirect to the user details. The delete button should be present and this will be done in issue #600. Both icons and user's name must have a hover behaviour.

A member only sees the user teams count and which user is an admin. 

## User story it relates to

#594 

### Basic example

image

### PR DESCRIPTION
**Pull request by RafaelSBatista97**
#599 

Relates to #594 

## Screenshots (if visual changes)
As Super Admin:
image

As member:
image

## Proposed Changes
  - As a super admin the switch button should be visible and editable. This action makes a request to the backend in order to change the user role in the app. 
  - The edit button and when the user clicks in the user's name should redirect to the user details. 
  - The delete button should be present and this will be done in issue https://github.com/xgeekshq/split/issues/600. 
  - Both icons and user's name must have a hover behaviour.
  - A member only sees the user teams count and which user is an admin.

Mention people who discussed this issue previously
@GuiSanto 

This pull request closes #599

### CODE REVIEW

File: backend/src/libs/guards/superAdmin.guard.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export class SuperAdminGuard implements CanActivate {
+	async canActivate(context: ExecutionContext) {
+		const request = context.switchToHttp().getRequest();
+
+		try {
+			return request.user.isSAdmin;
+		} catch (error) {
+			throw new ForbiddenException();
+		}
+	}
```
Comment (rpvsilva) [inline backend/src/libs/guards/superAdmin.guard.ts:13]: Can you confirm to me one thing? If you return `false`, without this `try { ... } catch () { ... }` and check what error it returns? If it returns a `ForbiddenException` you can remove this try catch and only return the `request.user.isSAdmin`

File: backend/src/modules/users/controller/users.controller.ts [hunk 2]
```diff
@@ -68,4 +79,40 @@ export default class UsersController {
...
+	})
+	@ApiInternalServerErrorResponse({
+		description: 'Internal Server Error',
+		type: InternalServerErrorResponse
+	})
+	@UseGuards(SuperAdminGuard)
+	@Put('/sadmin')
+	async updateUserSuperAdmin(@Body() userData: UpdateUserDto) {
+		return await this.updateUserApp.updateSuperAdmin(userData);
+	}
```
Comment (rpvsilva) [inline backend/src/modules/users/controller/users.controller.ts:117]:
```suggestion
	updateUserSuperAdmin(@Body() userData: UpdateUserDto) {
		return this.updateUserApp.updateSuperAdmin(userData);
	}
```

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: backend/src/libs/guards/superAdmin.guard.ts (hunk 1 above)
Comment (RafaelSBatista97) [reply backend/src/libs/guards/superAdmin.guard.ts]: yes, works fine without the try catch.

**Review by rpvsilva — APPROVED** [review]
LGTM 🔥

File: frontend/src/components/Users/UsersList/partials/CardUser/CardBody.tsx [hunk 3]
```diff
@@ -44,6 +48,16 @@ const CardBody = React.memo<CardBodyProps>(({ userWithTeams }) => {
...
 
+  let teamsSeparatedByComma: string[] = [];
+  if (teamsNames) {
+    teamsSeparatedByComma = teamsNames?.map((team, index) => {
+      if (index !== teamsNames.length - 1) {
+        return team.concat(', ');
+      }
+      return team;
+    });
+  }
```
Comment (rpvsilva) [inline frontend/src/components/Users/UsersList/partials/CardUser/CardBody.tsx:59]:
```suggestion
  const teamsSeparatedByComma = teamsNames?.join(', ') || '';
```

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: frontend/src/components/Users/UsersList/partials/CardUser/CardBody.tsx (hunk 3 above)
Comment (GuiSanto) [reply frontend/src/components/Users/UsersList/partials/CardUser/CardBody.tsx:59]: nice

**Review by rpvsilva — CHANGES_REQUESTED** [review]

**Review by rpvsilva — DISMISSED** [review]

File: frontend/src/components/Users/UsersList/partials/CardUser/SuperAdmin.tsx [hunk 4]
```diff
@@ -1,40 +1,87 @@
...
+      _id: userId,
+      isSAdmin: checked,
+    };
+
+    try {
+      await mutateAsync(updateTeamUser);
+      setCheckedState(checked);
+    } catch (error) {
+      setCheckedState(!checked);
+    }
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/partials/CardUser/SuperAdmin.tsx:35]: Is this working when something goes wrong? you have access to the request status

Comment (RafaelSBatista97) [reply frontend/src/components/Users/UsersList/partials/CardUser/SuperAdmin.tsx:35]: yes, it's working when something goes wrong. At least we tested it by putting a wrong url on the fetch and it goes to the catch showing the error.

**Review by rpvsilva — APPROVED** [review]

File: frontend/src/components/Users/UsersList/partials/CardUser/SuperAdmin.tsx [hunk 5]
```diff
@@ -1,40 +1,87 @@
...
+          <ConfigurationSettings
+            handleCheckedChange={handleSuperAdminChange}
+            isChecked={checkedState}
+            text=""
+            title="Super Admin"
+            disabled={isDisabledState}
+            styleVariant={isDisabledState}
+          />
+        )}
+      </Flex>
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/partials/CardUser/SuperAdmin.tsx:60]:
```suggestion
      <Flex css={{ ml: '$2', display: 'flex', alignItems: 'center' }}>
          <ConfigurationSettings
            handleCheckedChange={handleSuperAdminChange}
            isChecked={checkedState}
            text=""
            title="Super Admin"
            disabled={loggedUserId !== userId ? undefined : isDisabledState}
            styleVariant={loggedUserId !== userId ? undefined : isDisabledState}
          />
      </Flex>
```

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Pull request merged**